### PR TITLE
Cmake: Improved setconfig support for setting multiple parameters

### PR DESCRIPTION
**Pull request by simbit18**
## Summary
Improved setconfig support for setting multiple parameters for the .config

Added Cmake and Ninja version info

Added utility target to check .config from board's defconfig for refresh stage on GITHUB

## Impact
Impact on user: No changes to user-facing functionality
Impact on build: Build process remains the same
## Testing
We tested on the local platform MSYS2 and GitHub 

bin
version

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



The PR partially meets the NuttX requirements. Here's a breakdown:

**Strengths:**

* **Summary:** Provides a decent overview of the changes. 
* **Testing:**  Includes testing information (platform and evidence with screenshots).

**Areas for Improvement:**

* **Impact:**  The impact section needs to be more thorough.
    * **Impact on hardware:** Does this affect any specific architectures or boards? Be specific.
    * **Impact on documentation:**  If no documentation updates are needed, state that explicitly.  If updates are needed, provide them in the PR or a linked PR.
    * **Impact on security, compatibility:**  Even if the answer is NO, you must explicitly state it for each item.  Don't leave any blanks.
* **Testing:**
    * **Logs:** The screenshots provide visual confirmation but lack context. Include relevant snippets of build logs or test output to demonstrate the changes' effects. 
    * **Target(s):** Specify the exact target architectures and boards tested.  "Local platform MSYS2" is not specific enough for a NuttX target. 
    * **Before/After:** The description mentions testing on MSYS2 and GitHub, but it's unclear what was tested on each.  Provide distinct "before" and "after" logs for each testing environment (local and GitHub).

**Recommendation:**

Revise the PR to address the missing information, particularly in the "Impact" and "Testing" sections. Provide specific details about the testing environments, include relevant log snippets, and address all impact points explicitly.

File: tools/testbuild.sh [hunk 1]
```diff
@@ -450,8 +450,8 @@ function refresh_cmake {
     kconfig-tweak --file $nuttx/build/.config -d $toolchain
   fi
 
-  if ! cmake --build build -t savedefconfig 1>/dev/null; then
-    cmake --build build -t savedefconfig
+  if ! cmake --build build -t nuttx_job_refresh 1>/dev/null; then
```
Comment (xiaoxiang781216) [inline tools/testbuild.sh:453]: why not use savedefconfig or migrate the action in nuttx_job_refresh to savedefconfig?

Comment (simbit18) [reply tools/testbuild.sh:453]: Hi @xiaoxiang781216 
I created a simple test workflow with a NuttX repository with this PR by modifying the refresh_cmake function (testbuild.sh file) to cmake --build build -t savedefconfig.
There are these jobs sim-02 and arm-10 for Linux and macOS (even on macOS it must be possible to build NuttX with Cmake + Ninja)

https://github.com/simbit18/nuttx-testing-ci/actions/runs/11178303686/job/31075667531


linux
```
====================================================================================
Cmake in present: stm32f3discovery/usbnsh,CONFIG_ARM_TOOLCHAIN_CLANG
Configuration/Tool: stm32f3discovery/usbnsh,CONFIG_ARM_TOOLCHAIN_CLANG
2024-10-04 10:41:38
------------------------------------------------------------------------------------
  Cleaning...
  Configuring...
  Select HOST_LINUX=y
  Disabling CONFIG_ARM_TOOLCHAIN_GNU_EABI
  Enabling CONFIG_ARM_TOOLCHAIN_CLANG
  Building NuttX...
On branch simbit-savedefconfig
Your branch is up to date with 'origin/simbit-savedefconfig'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   boards/arm/stm32/stm32f3discovery/configs/usbnsh/defconfig

no changes added to commit (use "git add" and/or "git commit -a")
====================================================================================
```

macOS

```
====================================================================================
Cmake in present: stm32_tiny/nsh,CONFIG_ARM_TOOLCHAIN_CLANG
Configuration/Tool: stm32_tiny/nsh,CONFIG_ARM_TOOLCHAIN_CLANG
2024-10-04 10:34:54
------------------------------------------------------------------------------------
  Cleaning...
  Configuring...
  Select HOST_MACOS=y
  Disabling CONFIG_ARM_TOOLCHAIN_BUILDROOT
  Enabling CONFIG_ARM_TOOLCHAIN_CLANG
  Building NuttX...
On branch simbit-savedefconfig
Your branch is up to date with 'origin/simbit-savedefconfig'.

Changes not staged for commit:
	modified:   boards/arm/stm32/stm32_tiny/configs/nsh/defconfig

no changes added to commit
====================================================================================
```

nuttx_job_refresh behaves with the same logic as the tools/refresh.sh script used for the refresh stage (make build) on GitHub.

See https://github.com/apache/nuttx/blob/master/tools/refresh.sh

```
example nucleo-l152re/nsh

./tools/refresh.sh --silent nucleo-l152re/nsh
+ rm -f SAVEconfig
+ mv .config SAVEconfig
+ cp -a boards/arm/stm32/nucleo-l152re/configs/nsh/defconfig .config
+ make olddefconfig
+ make savedefconfig
+ diff boards/arm/stm32/nucleo-l152re/configs/nsh/defconfig defconfig
```

Comment (xiaoxiang781216) [reply tools/testbuild.sh:453]: so, we should replace savedefconfig cmake function with your implementation, instead adding a new nuttx_job_refresh.

**Review by xiaoxiang781216 — APPROVED** [review]

File: tools/testbuild.sh (hunk 1 above)
Comment (simbit18) [reply tools/testbuild.sh]: @xiaoxiang781216  done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**